### DISCUSSION TITLE
[FEATURE REQUEST] - Add array param to x-for

### DISCUSSION THREAD

**Discussion by go4cas** [alpinejs/alpine#768] [4. General]
I have a use case whereby I need to evaluate the current item, in a `x-for` loop, with the previous item.

The workaround at the moment is something like this:
```html
<template x-for="(item, index) in data.parentArray.childObject.grandChildArray" :key="index">
  <div :class="'bg-red-500': (index > 0 && item.someValue !== data.parentArray.childObject.grandChildArray[index-1].someValue)"></div>
</template>
```

I thought, seeing that `x-for` uses `forEach`, which supports the optional `array` parameter (providing access to the array object), it would be a good feature to have this available. So, turning the above example into:
```html
<template x-for="(item, index, array) in data.parentArray.childObject.grandChildArray" :key="index">
  <div :class="'bg-red-500': (index > 0 && item.someValue !== array[index-1].someValue)"></div>
</template>
```

**Comment by ryangjchandler** [comment]
@go4cas This is actually already possible, see the CodePen here: https://codepen.io/ryangjchandler/pen/KKzBKry

When you click the buttons and look in the console, you'll actually see a `Proxy` instead of a native `Array`, but this is just because Alpine wraps them as `Proxy` objects to allow for reactivity. They still behave like regular arrays.

**Reply by go4cas** [reply]
Oh, wow! I have just changed my code, and it is working out the box! Thanks, @ryangjchandler!

So, maybe this feature request should just be changed to an update to the `README`?

**Reply by ryangjchandler** [reply]
Feel free to send in a PR for it! 🤗
